### ISSUE TITLE
Avg IOU: -nan, why?

### ISSUE THREAD

**Issue by jiaswee1995** [pjreddie/darknet#746]
![image](https://user-images.githubusercontent.com/22536079/39398864-12c75f86-4b47-11e8-8270-38016b104f9e.png)

**Comment by Johnccl** [comment]
In the source code, you can find Avg IOU = avg_iou / count, so if count is 0, Avg IOU is nan, and so does the rest.

**Comment by sunshinezhihuo** [comment]
I also have the nan problem. I just want to train the network for person detection. Then I set classes 2 (background + person) and use VOC data format. And I only have people datasets. Nan always occurs when I train this network.  Should I ignore this problem? @jiaswee1995 @Johnccl

**Comment by Johnccl** [comment]
You can see the answer of AlexeyAB about nan: https://github.com/AlexeyAB/darknet/issues/636#issuecomment-381400954.
When I train yolov3 on traffic sign datasets, nan always occurs but the model works well finally. @sunshinezhihuo

**Comment by sunshinezhihuo** [comment]
@Johnccl Ok, thank you very much.  In yolov3.cfg, I set:
`batch =128
subdivisions =64
learning_rate = 0.0001`. 
 When I train the network, the loss decreases at the beginning.  But  after some iterations, the loss becomes increased. Should I change the  learning_rate more smaller?

**Comment by Johnccl** [comment]
The fluctuation of loss is normal, maybe you just need to train for more iterations(e.g., 40000).
But if the loss increases all the time, you may check your code. @sunshinezhihuo

**Comment by sunshinezhihuo** [comment]
OK, thank you @Johnccl
